### PR TITLE
Weptech gateway

### PR DESCRIPTION
**Pull request by thomas-blank**
This is just a dummy pull request for our internal reviewing process. Once we're done, we'll post the actual PR to the official contiki repository.

### CODE REVIEW

File: examples/cc2538-cc1200-eth/ip64-rpl-border-router/Makefile [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  PROJECT_SOURCEFILES += httpd-simple.c httpd-simple-handler.c
+else ifneq ($(WITH_WEBSERVER),0)
+  APPS += $(WITH_WEBSERVER)
+  DEFINES += UIP_CONF_TCP=1
+  DEFINES += WEBSERVER=2
+endif
+
+### Use IPv6 stack
+CONTIKI_WITH_IPV6 = 1
+
```
Comment (thomas-blank) [inline examples/cc2538-cc1200-eth/ip64-rpl-border-router/Makefile]: Check whether we can add rimestats to have some physical layer debugging stats for the CC1200.

File: examples/cc2538-cc1200-eth/ip64-rpl-border-router/border-router.c [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+      print_stats();
+#endif /* BLINK_INTERVAL */
+    }
+
+  }
+
+  PROCESS_END();
+
+}
+/*---------------------------------------------------------------------------*/
```
Comment (thomas-blank) [inline examples/cc2538-cc1200-eth/ip64-rpl-border-router/border-router.c]: Please don't check in binary files.

File: examples/cc2538-cc1200-eth/ip64-rpl-border-router/project-conf.h [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+#endif
+
+/* Size of the buffer for the html page. */
+#define BUF_SIZE                        256
+/*---------------------------------------------------------------------------*/
+/* Configures the webserver under apps/webserver */
+#define WEBSERVER_CONF_STATUSPAGE       1
+#define WEBSERVER_CONF_CFS_CONNS        2
+/*---------------------------------------------------------------------------*/
+#define UIP_CONF_RECEIVE_WINDOW         128
```
Comment (thomas-blank) [inline examples/cc2538-cc1200-eth/ip64-rpl-border-router/project-conf.h]: Comment why.

File: examples/sensniff/cc2538-cc1200-eth/target-conf.h [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+/* Enable sniff mode for cc2538 2.4 GHz radio  */
+#define CC2538_RF_CONF_SNIFFER          1
+#define CC2538_RF_CONF_AUTOACK          0
+#endif /* PLATFORM_CONF_USE_CC1200 */
+/*---------------------------------------------------------------------------*/
+#define UART0_CONF_BAUD_RATE            460800
+/*---------------------------------------------------------------------------*/
+
+#define SENSNIFF_IO_DRIVER_H            "pool/cc2538-io.h"
+#endif /* TARGET_CONF_H_ */
```
Comment (thomas-blank) [inline examples/sensniff/cc2538-cc1200-eth/target-conf.h:53]: Binaries again!

File: platform/cc2538-cc1200-eth/Makefile.cc2538-cc1200-eth [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+endif
+
+BSL = $(CONTIKI)/tools/cc2538-bsl/cc2538-bsl.py -b 460800
+
+%.upload: %.bin
+ifeq ($(wildcard $(BSL)), )
+	@echo "ERROR: Could not find the cc2538-bsl script. Did you run 'git submodule update --init' ?"
+else
+	python $(BSL) -e -w -v $<
+endif
```
Comment (thomas-blank) [inline platform/cc2538-cc1200-eth/Makefile.cc2538-cc1200-eth]: Test  this

File: platform/cc2538-cc1200-eth/contiki-conf.h [hunk 6]
```diff
@@ -0,0 +1,755 @@
...
+
+#ifndef DBG_CONF_USB
+#define DBG_CONF_USB                0 /**< All debugging over UART by default */
+#endif
+
+#ifndef SERIAL_LINE_CONF_UART
+#define SERIAL_LINE_CONF_UART       0 /**< UART to use with serial line */
+#endif
+
+#if !SLIP_ARCH_CONF_USB
```
Comment (thomas-blank) [inline platform/cc2538-cc1200-eth/contiki-conf.h]: Try and make Slip work.

File: platform/cc2538-cc1200-eth/contiki-conf.h [hunk 7]
```diff
@@ -0,0 +1,755 @@
...
+/**
+ * \name Radio selection
+ *
+ * The main CC2538 processor drives the 2.4GHz antenna, for Sub-GHz operation we
+ * use an additional CC1200 chip that the CC2538 controls via SPI. Only one of
+ * these configurations can be operated at a time. To enable the CC1200 and
+ * disable the CC2538 radio core, set PLATFORM_CONF_USE_CC1200 to 1.
+ * @{
+ */
+#if CC1200_RF_TESTMODE
```
Comment (thomas-blank) [inline platform/cc2538-cc1200-eth/contiki-conf.h]: Comment on these.

File: platform/cc2538-cc1200-eth/contiki-conf.h [hunk 8]
```diff
@@ -0,0 +1,755 @@
...
+/*---------------------------------------------------------------------------*/
+#if PLATFORM_USE_CC1200
+#define NETSTACK_CONF_RADIO             cc1200_driver
+#else /* PLATFORM_USE_CC1200 */
+#define NETSTACK_CONF_RADIO             cc2538_rf_driver
+#endif /* PLATFORM_USE_CC1200 */
+/** @} */
+/*---------------------------------------------------------------------------*/
+/**
+ *  \name RF configuration for the CC120 radio chip
```
Comment (thomas-blank) [inline platform/cc2538-cc1200-eth/contiki-conf.h]: *cc1200

File: platform/cc2538-cc1200-eth/contiki-conf.h [hunk 9]
```diff
@@ -0,0 +1,755 @@
...
+ * @{
+ */
+/*---------------------------------------------------------------------------*/
+/**
+ * \brief Frequency fine-tuning.
+ *
+ * Please refer to equations 27 and 28 in the user guide and default values in
+ * the 802154g configuration file of the CC12000.
```
Comment (thomas-blank) [inline platform/cc2538-cc1200-eth/contiki-conf.h]: Just move the one up that is left over here.

File: platform/cc2538-cc1200-eth/contiki-conf.h [hunk 10]
```diff
@@ -0,0 +1,755 @@
...
+ * Please don't touch.
+ */
+#define CC1200_CONF_FREQ_OFFSET         691
+
+#ifndef CC1200_CONF_USE_GPIO2
+#define CC1200_CONF_USE_GPIO2           1
+#endif /* CC1200_CONF_USE_GPIO2 */
+
+#ifndef CC1200_CONF_MAX_PAYLOAD_LEN
+//#define CC1200_CONF_MAX_PAYLOAD_LEN     127
```
Comment (thomas-blank) [inline platform/cc2538-cc1200-eth/contiki-conf.h]: Can this go? Research.

File: platform/cc2538-cc1200-eth/contiki-conf.h [hunk 11]
```diff
@@ -0,0 +1,755 @@
...
+#define RPL_CONF_MAX_DAG_ENTRIES                1
+#ifndef RPL_CONF_OF
+#define RPL_CONF_OF                             rpl_mrhof
+#endif
+
+#define UIP_CONF_ND6_REACHABLE_TIME             600000
+#define UIP_CONF_ND6_RETRANS_TIMER              10000
+
+#ifndef NBR_TABLE_CONF_MAX_NEIGHBORS
+#define NBR_TABLE_CONF_MAX_NEIGHBORS            20
```
Comment (thomas-blank) [inline platform/cc2538-cc1200-eth/contiki-conf.h]: Wasn't this 32 at one point?

File: platform/cc2538-cc1200-eth/contiki-conf.h [hunk 12]
```diff
@@ -0,0 +1,755 @@
...
+/* Define our IPv6 prefixes/contexts here */
+#define SICSLOWPAN_CONF_MAX_ADDR_CONTEXTS       1
+#ifndef SICSLOWPAN_CONF_ADDR_CONTEXT_0
+#define SICSLOWPAN_CONF_ADDR_CONTEXT_0 {                   \
+    addr_contexts[0].prefix[0] = UIP_DS6_DEFAULT_PREFIX_0; \
+    addr_contexts[0].prefix[1] = UIP_DS6_DEFAULT_PREFIX_1; \
+  }
+#endif
+
+#define MAC_CONF_CHANNEL_CHECK_RATE             8
```
Comment (thomas-blank) [inline platform/cc2538-cc1200-eth/contiki-conf.h]: Move this to RDC

File: platform/cc2538-cc1200-eth/contiki-main.c [hunk 13]
```diff
@@ -0,0 +1,303 @@
...
+  netstack_init();
+
+#if NETSTACK_CONF_WITH_IPV6
+  memcpy(&uip_lladdr.addr, &linkaddr_node_addr, sizeof(uip_lladdr.addr));
+  queuebuf_init();
+  process_start(&tcpip_process, NULL);
+#endif /* NETSTACK_CONF_WITH_IPV6 */
+
+  energest_init();
+  ENERGEST_ON(ENERGEST_TYPE_CPU);
```
Comment (thomas-blank) [inline platform/cc2538-cc1200-eth/contiki-main.c]: Energest: Is this correct, where are the leds?

File: platform/cc2538-cc1200-eth/dev/cc1200-arch.c [hunk 14]
```diff
@@ -0,0 +1,333 @@
...
+#define CC1200_GPIO2_PORT_VECTOR    GPIO_C_IRQn
+#elif (CC1200_GPIO2_PORT == GPIO_D_NUM)
+#define CC1200_GPIO2_PORT_VECTOR    GPIO_D_IRQn
+#endif
+/*---------------------------------------------------------------------------*/
+static void
+gpiox_callback(uint8_t port, uint8_t pin)
+{
+
+  /*
```
Comment (thomas-blank) [inline platform/cc2538-cc1200-eth/dev/cc1200-arch.c]: this can go.

File: platform/cc2538-cc1200-eth/dev/cc1200-arch.c [hunk 15]
```diff
@@ -0,0 +1,333 @@
...
+   * Call-stack up to here
+   * - gpio_port_b_isr() (gpio.c)
+   * - notify() (gpio.c)
+   *
+   * ENERGEST_ON / OFF + lpm_exit() + interrupt trigger reset is performed
+   * within gpio_port_x_isr()
+   */
+
+  leds_on(LEDS_RED);
```
Comment (thomas-blank) [inline platform/cc2538-cc1200-eth/dev/cc1200-arch.c]: Remove

File: platform/cc2538-cc1200-eth/dev/enc28j60-mac-addr-provider.c [hunk 16]
```diff
@@ -0,0 +1,67 @@
...
+ *
+ */
+
+#include <stddef.h>
+#include <stdint.h>
+
+#include "dev/enc28j60/enc28j60-ip64-driver.h"
+#include "dev/eui48.h"
+
+static uint8_t eui48[EUI48_SIZE];
```
Comment (thomas-blank) [inline platform/cc2538-cc1200-eth/dev/enc28j60-mac-addr-provider.c]: Remove this, add a pointer parameter to the function.

File: platform/cc2538-cc1200-eth/dev/i2c-eeprom.c [hunk 17]
```diff
@@ -0,0 +1,202 @@
...
+           */
+          i2c_master_command(I2C_MASTER_CMD_BURST_SEND_FINISH);
+          while (i2c_master_busy());
+          break;
+        }
+
+      } /* for(;;) */
+
+#if 1
+      /* Poll ACK */
```
Comment (thomas-blank) [inline platform/cc2538-cc1200-eth/dev/i2c-eeprom.c]: Wait for ack. Remove the dead branch.

File: platform/cc2538-cc1200-eth/README.md [hunk 18]
```diff
@@ -0,0 +1,504 @@
...
+* GNU Tools for ARM Embedded Processors. This is the recommended version.
+  <https://launchpad.net/gcc-arm-embedded>
+* Alternatively, you can use this older version for Linux. At the time of 
+  writing, this is the version used by Contiki's regression tests. 
+  <https://sourcery.mentor.com/public/gnu_toolchain/arm-none-eabi/arm-2008q3-66-arm-none-eabi-i686-pc-linux-gnu.tar.bz2>
+
+Drivers
+-------
+On Windows, the drivers for the FTDI234 USB chip on the gateway will be
+installed along with [smart-rf-flashprog](TI Flash Programmer 2).
```
Comment (uknoblic) [inline platform/cc2538-cc1200-eth/README.md]: See FTDI web page (http://www.ftdichip.com/FTDrivers.htm)

File: platform/cc2538-cc1200-eth/README.md [hunk 19]
```diff
@@ -0,0 +1,504 @@
...
+
+
+Requirements
+============
+To get started with the Gateway, you will need the following:
+
+ * A toolchain to compile Contiki for the CC2538.
+ * Maybe additional drivers so that your OS can communicate with your hardware.
+ * Software to upload images to the CC2538.
+ * Ideally some more 6LoWPan hardware to have a mesh for which the Gateway can
```
Comment (uknoblic) [inline platform/cc2538-cc1200-eth/README.md]: 6LoWPAN (Typo)

File: platform/cc2538-cc1200-eth/README.md [hunk 20]
```diff
@@ -0,0 +1,504 @@
...
+Examples
+========
+Under `examples/cc2538-cc1200-eth` you will find the following examples.
+
+ip64-rpl-border-router
+----------------------
+This example will have the Gateway act as a border router between a 6LoWPAN
+wireless network and a IPv4 ethernet network with internet access, allowing the
+wireless devices to communicate with any internet server. The example we will
+use on the client device will send senosr data via MQTT to a pre-configured
```
Comment (uknoblic) [inline platform/cc2538-cc1200-eth/README.md]: Typo (senosr)

File: examples/cc2538-cc1200-eth/ip64-rpl-border-router/Makefile (hunk 1 above)
Comment (thomas-blank) [reply examples/cc2538-cc1200-eth/ip64-rpl-border-router/Makefile]: Done.

File: examples/cc2538-cc1200-eth/ip64-rpl-border-router/border-router.c (hunk 2 above)
Comment (thomas-blank) [reply examples/cc2538-cc1200-eth/ip64-rpl-border-router/border-router.c]: Done.

File: examples/cc2538-cc1200-eth/ip64-rpl-border-router/project-conf.h (hunk 3 above)
Comment (thomas-blank) [reply examples/cc2538-cc1200-eth/ip64-rpl-border-router/project-conf.h]: Done.

File: examples/sensniff/cc2538-cc1200-eth/target-conf.h (hunk 4 above)
Comment (thomas-blank) [reply examples/sensniff/cc2538-cc1200-eth/target-conf.h:53]: Done.

File: platform/cc2538-cc1200-eth/Makefile.cc2538-cc1200-eth (hunk 5 above)
Comment (thomas-blank) [reply platform/cc2538-cc1200-eth/Makefile.cc2538-cc1200-eth]: Done.

File: platform/cc2538-cc1200-eth/README.md (hunk 19 above)
Comment (thomas-blank) [reply platform/cc2538-cc1200-eth/README.md]: Done.

File: platform/cc2538-cc1200-eth/README.md (hunk 18 above)
Comment (thomas-blank) [reply platform/cc2538-cc1200-eth/README.md]: Done.

File: platform/cc2538-cc1200-eth/README.md (hunk 20 above)
Comment (thomas-blank) [reply platform/cc2538-cc1200-eth/README.md]: Done.

File: platform/cc2538-cc1200-eth/contiki-conf.h (hunk 6 above)
Comment (thomas-blank) [reply platform/cc2538-cc1200-eth/contiki-conf.h]: Done.

File: platform/cc2538-cc1200-eth/contiki-conf.h (hunk 7 above)
Comment (thomas-blank) [reply platform/cc2538-cc1200-eth/contiki-conf.h]: Done.

File: platform/cc2538-cc1200-eth/contiki-conf.h (hunk 8 above)
Comment (thomas-blank) [reply platform/cc2538-cc1200-eth/contiki-conf.h]: Done.

File: platform/cc2538-cc1200-eth/contiki-conf.h (hunk 9 above)
Comment (thomas-blank) [reply platform/cc2538-cc1200-eth/contiki-conf.h]: I took the one in the middle.

File: platform/cc2538-cc1200-eth/contiki-conf.h (hunk 10 above)
Comment (thomas-blank) [reply platform/cc2538-cc1200-eth/contiki-conf.h]: Done.

File: platform/cc2538-cc1200-eth/contiki-conf.h (hunk 11 above)
Comment (thomas-blank) [reply platform/cc2538-cc1200-eth/contiki-conf.h]: Done.

File: platform/cc2538-cc1200-eth/contiki-conf.h (hunk 12 above)
Comment (thomas-blank) [reply platform/cc2538-cc1200-eth/contiki-conf.h]: Was dead, removed.

File: platform/cc2538-cc1200-eth/contiki-main.c (hunk 13 above)
Comment (thomas-blank) [reply platform/cc2538-cc1200-eth/contiki-main.c]: That is all done in the respective modules. To remove the ambiguity, I moved the green LED fade before this line. The startup is just not energested.

File: platform/cc2538-cc1200-eth/dev/cc1200-arch.c (hunk 14 above)
Comment (thomas-blank) [reply platform/cc2538-cc1200-eth/dev/cc1200-arch.c]: Done.

File: platform/cc2538-cc1200-eth/dev/cc1200-arch.c (hunk 15 above)
Comment (thomas-blank) [reply platform/cc2538-cc1200-eth/dev/cc1200-arch.c]: Done

File: platform/cc2538-cc1200-eth/dev/enc28j60-mac-addr-provider.c (hunk 16 above)
Comment (thomas-blank) [reply platform/cc2538-cc1200-eth/dev/enc28j60-mac-addr-provider.c]: Done.

File: platform/cc2538-cc1200-eth/dev/i2c-eeprom.c (hunk 17 above)
Comment (thomas-blank) [reply platform/cc2538-cc1200-eth/dev/i2c-eeprom.c]: Done.

**Pull request closed without merging**